### DISCUSSION TITLE
Undo/redo middleware

### DISCUSSION THREAD

**Discussion by KrisLau** [pmndrs/zustand#1611] [Ideas] [answered]
Just found this today when I was trying to figure out how to implement undo and redo: https://github.com/charkour/zundo. Seems to be in beta but I thought it might make a good addition to the middleware section of the docs

**Comment by KrisLau** [comment, marked as answer]
JK just saw it's already in the third party libraries doc: https://github.com/pmndrs/zustand/blob/3788fe3652ad624c1d85d91b52ded53320f14bc3/docs/integrations/third-party-libraries.md

**Comment by charkour** [comment]
Zundo is out of beta now! Thank you :)

**Reply by KrisLau** [reply]
It's a fantastic package!

**Reply by charkour** [reply]
Thanks for the kind words :)

**Reply by ohthepain** [reply]
Does zundo support undo across multiple slices?

**Reply by charkour** [reply]
Yes, I believe it should. If you get it working, please submit a pull request using an example!
